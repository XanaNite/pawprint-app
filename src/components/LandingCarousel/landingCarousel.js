import React, { Component } from 'react';
import "react-responsive-carousel/lib/styles/carousel.min.css";
import { Carousel } from 'react-responsive-carousel';
import derp from '../../assets/images/derp.jpg';
import puggy from '../../assets/images/puggy.jpg';
import smartpup from '../../assets/images/smartpup.jpg';

export default class LandingCarousel extends Component {
    render() {
        return (
            <div className='carousel'>
            <Carousel autoPlay>
                    <div>
                        <img src={derp} />
                        <p className="legend">Snickers</p>
                    </div>
                    <div>
                        <img src={smartpup} />
                        <p className="legend">Janet</p>
                    </div>
                    <div>
                        <img src={puggy} />
                        <p className="legend">Elvis</p>
                    </div>
    
            </Carousel>
            </div>
        );
    }
};

