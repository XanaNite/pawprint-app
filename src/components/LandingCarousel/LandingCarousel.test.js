import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LandingCarousel from './LandingCarousel';

describe(`LandingCarousel component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders a form.LandingCarousel by default', () => {
    const wrapper = shallow(<LandingCarousel />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the LandingCarousel given props', () => {
    const wrapper = shallow(<LandingCarousel {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
