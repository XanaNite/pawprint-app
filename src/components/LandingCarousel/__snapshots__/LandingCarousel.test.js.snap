// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingCarousel component renders a form.LandingCarousel by default 1`] = `
<div
  className="carousel"
>
  <Carousel
    autoPlay={true}
    axis="horizontal"
    centerMode={false}
    centerSlidePercentage={80}
    dynamicHeight={false}
    emulateTouch={false}
    infiniteLoop={false}
    interval={3000}
    labels={
      Object {
        "item": "slide item",
        "leftArrow": "previous slide / item",
        "rightArrow": "next slide / item",
      }
    }
    onChange={[Function]}
    onClickItem={[Function]}
    onClickThumb={[Function]}
    onSwipeEnd={[Function]}
    onSwipeMove={[Function]}
    onSwipeStart={[Function]}
    selectedItem={0}
    showArrows={true}
    showIndicators={true}
    showStatus={true}
    showThumbs={true}
    statusFormatter={[Function]}
    stopOnHover={true}
    swipeScrollTolerance={5}
    swipeable={true}
    transitionTime={350}
    useKeyboardArrows={false}
    verticalSwipe="standard"
    width="100%"
  >
    <div>
      <img
        alt=""
        src="derp.jpg"
      />
      <p
        className="legend"
      >
        Snickers
      </p>
    </div>
    <div>
      <img
        alt=""
        src="smartpup.jpg"
      />
      <p
        className="legend"
      >
        Janet
      </p>
    </div>
    <div>
      <img
        alt=""
        src="puggy.jpg"
      />
      <p
        className="legend"
      >
        Elvis
      </p>
    </div>
  </Carousel>
</div>
`;

exports[`LandingCarousel component renders the LandingCarousel given props 1`] = `
<div
  className="carousel"
>
  <Carousel
    autoPlay={true}
    axis="horizontal"
    centerMode={false}
    centerSlidePercentage={80}
    dynamicHeight={false}
    emulateTouch={false}
    infiniteLoop={false}
    interval={3000}
    labels={
      Object {
        "item": "slide item",
        "leftArrow": "previous slide / item",
        "rightArrow": "next slide / item",
      }
    }
    onChange={[Function]}
    onClickItem={[Function]}
    onClickThumb={[Function]}
    onSwipeEnd={[Function]}
    onSwipeMove={[Function]}
    onSwipeStart={[Function]}
    selectedItem={0}
    showArrows={true}
    showIndicators={true}
    showStatus={true}
    showThumbs={true}
    statusFormatter={[Function]}
    stopOnHover={true}
    swipeScrollTolerance={5}
    swipeable={true}
    transitionTime={350}
    useKeyboardArrows={false}
    verticalSwipe="standard"
    width="100%"
  >
    <div>
      <img
        alt=""
        src="derp.jpg"
      />
      <p
        className="legend"
      >
        Snickers
      </p>
    </div>
    <div>
      <img
        alt=""
        src="smartpup.jpg"
      />
      <p
        className="legend"
      >
        Janet
      </p>
    </div>
    <div>
      <img
        alt=""
        src="puggy.jpg"
      />
      <p
        className="legend"
      >
        Elvis
      </p>
    </div>
  </Carousel>
</div>
`;
